fs/vfs/fs_read.c

```diff
@@ -102,6 +101,11 @@ static ssize_t file_readv_compat(FAR struct file *filep,
       remaining -= nread;
     }
 
+  if (ntotal >= 0)
+    {
+      uio_advance(uio, ntotal);
```

Comment:
what's benefit to update the offset in one of uio entry

Reply:
and maybe can help things like restart on EINTR in future.

Reply:
can we advance in the loop to avoid iterate the vector twice

Reply:
yes, we can. i feel it's one of the last things to optimize though.